### DISCUSSION TITLE
Is it possible to have subfolders in the theme folder?

### DISCUSSION THREAD

**Discussion by PetersOtto** [datenstrom/community#928] [Ask a question] [answered]
I would like to use subfolders in the theme folder. Is that somehow possible?
Perhaps the following structure:

```
├── content               
│   ├── 1-home            
│   ├── 9-about           
│   └── shared            
├── media                 
│   ├── downloads         
│   ├── images            
│   └── thumbnails        
└── system                
    ├── extensions 
    ├── layouts           
    ├── themes
    │   └── myTheme
    │       ├── css
    │       ├── favicon
    │       ├── fonts
    │       ├── js
    │       └── scss 
    └── trash
```

**Comment by pftnhr** [comment]
As far as I understand it, at least the `theme.css` and the `theme.png` must be in the theme directory. At least that's how it works for me.

**Comment by annaesvensson** [comment, marked as answer]
As Giovanni probably would say, It's easier than you imagine. 😄 The core checks that a file like `system/themes/mytheme.css` exists and the rest is up to you. If you want to include additional CSS, JavaScript and image files from subfolders, you can do this the usual way. Either add links to your `system/layouts/header.html` file or do this automatically with the help of an extension.

**Comment by PetersOtto** [comment]
Thank you both!

I am currently tidying up my code. I was wondering if it might be better to put all theme files in the theme folder.
If I have understood you correctly I don't think I need to change anything then.
I've been working with the following folder structure for a long time and it works well for me:

```
├── content               
│   ├── 1-home            
│   ├── 9-about           
│   └── shared            
├── media                 
│   ├── downloads         
│   ├── images            
│   └── thumbnails        
└── system                
    ├── extensions
    │   └── myTheme
    │        ├── css
    │        ├── favicon
    │        ├── fonts
    │        ├── js
    │        └── scss    
    ├── layouts           
    ├── themes
    │   └── myTheme.css (empty)     
    └── trash  
```

**Reply by pftnhr** [reply]
You only need to adjust the paths in your `/system/extensions/theme.php`. And if you do an `@import` in `/system/themes/themes.css`, then there too.

**Reply by PetersOtto** [reply]
Hi Robert, `@import `rule is clear but I don't understand why I should customize the paths in the `theme.php` &#128556;

**Reply by pftnhr** [reply]
I meant stuf like that:

``` php
// Handle page extra data
    public function onParsePageExtra($page, $name) {
        $output = null;
        if ($name=="header") {
            $themeLocation = $this->yellow->system->get("coreServerBase").$this->yellow->system->get("CoreThemeLocation");
            $output = "<script defer=\"defer\" src=\"{$themeLocation}myTheme/js/myTheme.js\"></script>\n";
        }
        return $output;
    }
```

**Reply by PetersOtto** [reply]
Oh yes, of course! I've always written the links to the `js` and `css` files directly into my `header.html` or `footer.html`.
